src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,143 @@
+package duke.ui;
+
+import duke.exception.DukeException;
+import duke.task.TaskList;
+import duke.task.Task;
+
+import java.util.Scanner;
+
+/**
+ * Represents the user interface of the Duke chatbot.
+ */
+public class Ui {
+
+    /**
+     * Displays a welcome message to the user.
+     */
+    public void showWelcome() {
+        System.out.println("Hello! I'm Bob");
+        System.out.println("What can I do for you?");
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Reads a command entered by the user.
+     *
+     * @return The command entered by the user.
+     */
+    public String readCommand() {
+        Scanner scanner = new Scanner(System.in);
+        return scanner.nextLine();
+    }
+
+    /**
+     * Displays a horizontal line separator.
+     */
+    public void showLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Displays an error message to the user.
+     *
+     * @param errorMessage The error message to be displayed.
+     */
+    public void showError(String errorMessage) {
+        System.out.println("☹ " + errorMessage);
```

Comment:
I like the inclusion of a unicode face. I wonder if that will cause a potential compatibility issue, though.